### PR TITLE
drivers: ams-tmd2635: add driver code for proximitory sensor tmd2635

### PR DESCRIPTION
**Pull request by yunheQin**
This PR adds basic support of proximitory sensor tmd2635. The sensor was tested with MIMXRT1062xxx6A for monitoring position value of sample with following output.

Untitled picture

### CODE REVIEW

File: drivers/can/can_shell.c
Comment (martinjaeger) [inline drivers/can/can_shell.c:1]: This change seems unrelated to the title of this PR. Please split into separate PR.

File: drivers/sensor/tmd2635/tmd2635.c [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+}
+
+static int tmd2635_channel_get(const struct device *dev, enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct tmd2635_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_PROX) {
+		/* inverting sensor data to fit Zephyr */
+		// LOG_INF("original data: %X ..",data->pdata);
```
Comment (martinjaeger) [inline drivers/sensor/tmd2635/tmd2635.c:164]: Use C90 style comments only and no dead (commented-out) code, please.

File: drivers/sensor/tmd2635/tmd2635.c [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+		LOG_ERR("Failed reading calibra status");
+		return ret;
+	}
+	// else{
+	// 		if (calibra_status & TMD2635_CALIBSTAT_CALIB_FINISHED )
+	// 		{
+	// 			status
+	// 		}
+
+	// }
```
Comment (martinjaeger) [inline drivers/sensor/tmd2635/tmd2635.c:204]: No dead code. Please remove.

File: samples/subsys/shell/shell_module/prj.conf
Comment (martinjaeger) [inline samples/subsys/shell/shell_module/prj.conf:1]: Unrelated changes for this PR.

**Review by martinjaeger — CHANGES_REQUESTED** [review]
Dear @yunheQin,

thanks for your contribution.

The changes to the shell seem to be unrelated to this PR and should be split into a dedicated PR.

Please have a look at the link below for commit message guidelines. The are almost correct, but we don't use feat/fix/... prefixes. 
https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines

See the additional comments for some first feedback.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
-1 on the CAN shell changes in this PR.

**Comment by FlorianGuhl-zeiss** [comment]
@yunheQin can you please remove @dipakgmx and my commits from this PR?

**Comment by martinjaeger** [comment]
@yunheQin Please don't open a new PR (if that is your plan), but fix this one instead. If you need help with git, just let us know.

**Comment by yunheQin** [comment]
All issues above have been solved, so please Reopen this PR.

**Comment by yunheQin** [comment]
The above all issues have been solved, and this PR should be opened to finish the code update.

**Comment by henrikbrixandersen** [comment]
> The above all issues have been solved, and this PR should be opened to finish the code update.

This PR cannot be reopened since the branch was either force-pushed or recreated while the PR was closed.

**Comment by martinjaeger** [comment]
Well, that's unfortunate, but in that case you'll have to open a new PR @yunheQin.

I had a look into your branch and there is still a merge commit in it: https://github.com/yunheQin/zephyr/commits/feature/add-proximitory-sensor-tmd2635

Please remove that before opening the new PR. As stated in the docs linked above, merge commits are not accepted.
